api/src/backend/api/v1/views.py

```diff
@@ -2460,6 +2462,116 @@ def metadata_latest(self, request):
         serializer.is_valid(raise_exception=True)
         return Response(serializer.data)
 
+    @action(
+        detail=True,
+        methods=["get"],
+        url_name="timeline",
+        url_path=r"timeline(?:/(?P<lookback_days>[0-9]+))?",
+    )
+    def timeline(self, request, pk=None, lookback_days=None):
```

Comment:
@MrCloudSec based on your tests, how long can this take?

@AdriiiPRodri @vicferpoy @josemazo @rakan-pro I'm not sure about having a synchronous endpoint to retrieve this data. From a design standpoint I think we'd have an async endpoint to enqueue a task to fetch that data. Then, another endpoint to get the status/result. What do you think?